File: tests/matrix/storage_folding/matrix_storage_folding_basic_doc.cpp
#include <optitrust.h>

int main() {
  int* const a = MALLOC1(int, 10);
  int* const b = MALLOC1(int, 8);
  for (int i = 0; i < 10; i++) {
     a[MINDEX1(10, i)] = i;
    if (i >= 2) {
      b[MINDEX1(8, i - 2)] = a[MINDEX1(10, i - 2)] +
                             a[MINDEX1(10, i - 1)] +
                             a[MINDEX1(10, i)];
    }
  }

  return 0;
}

File: tests/matrix/storage_folding/matrix_storage_folding_basic_doc.ml
open Optitrust
open Prelude
open Target

let _ = Run.script_cpp (fun _ ->
  let a = find_var "a" [] in
  !! Matrix_basic.storage_folding ~var:a ~dim:0 ~size:(trm_int 3) [cFunBody "main"];
)
